source/server/hot_restart_impl.cc

```diff
@@ -105,18 +112,25 @@ std::string SharedMemory::version(size_t max_num_stats, size_t max_stat_name_len
 }
 
 std::string SharedMemory::version() {
-  return version(num_stats_, Stats::RawStatData::maxNameLength());
+  return version(max_stats_, Stats::RawStatData::maxNameLength());
```

Comment:
Sorry, I omitted all the important context.

There's a command-line option to get the shmem version from an envoy binary, and an http admin command to get the shmem version from the running envoy.

Typically, the envoy startup script will get both and compare the two.  If they don't match, it will not try to do a hot-restart.

So if the startup script is working properly (and envoy is feeding it the right info), the runtime check in envoy will never fail.

Reply:
Ah, thanks.  That should be human-readable?  Looks like the current impl is.  Maybe hash "the quick brown fox..." instead of all the characters from 1-255?

Reply:
It should be clean ascii, but I don't think it needs to be human readable.  The only real requirement is that the strings compare unequal when it changes.

Reply:
I have this working but I wonder how you feel about HotRestartImplTest.versionString.  It seems to me like the it's just a kind of duplicate of the code.

WDYT of structuring the test as:
  construct hot_restart1.
  construct hot_restart2
  EXPECT_EQ(hot_restart1.version(), hot_restart2.version());
  tweek an option
  construct hot_restart3
  EXPECT_NE(hot_restart1.version(), hot_restart3.version());

Reply:
I don't know why the test is that way.  I like your suggestion here better than the current test, so I say go for it.

Reply:
done.